### ISSUE TITLE
dataset.title.sufficient

### ISSUE THREAD

**Issue by gothub** [NCEAS/metadig-checks#85]
## Description

Check if the dataset title has an appropriate length and contains needed information

## Priority

Choose a priority for the FAIR suite (Required or Optional)

- FAIR: Required

## Issues
- need to agree on and develop/obtain an algorithm to determine title `who`, `what`, `where`, `when`

## Procedure

- extract the title and perform a word count - if it is too long or too short, the check fails
- analyze the title to check for who, what, where, when (who is the creator, where were the resources collected or created, when did this happen)

**Comment by JEDamerow** [comment]
analysis of title should include topic (what), location (where), dates (when). The "who"/creator would not be included in the title.

**Comment by JEDamerow** [comment]
The current check in ESS-DIVE and Arctic Data Center is for between 7-20 words. Should include the exact number of words to look for here to be clear. Will it be the same here?

**Comment by mbjones** [comment]
Agreed @JEDamerow -- I added the word count to the description, and let's discuss here what the appropriate checks are.  We've had some feedback that 20 words is too low for the max.  In general, the title should be conveniently citable and recognizable as a title. More extended details should go in the abstract.  A hard limit is of course arbitrary. We know that short titles are generally inadequate to describe the content and context of datasets.  Maybe longer titles are acceptable?  Would a 400 word title be acceptable?  200? 100? 50? 30? 20?

**Comment by gothub** [comment]
In order to check the title for "what", "when", "where", this check could extract other fields in the metadata and check that they are in the title. For example, in EML:
- for when: extract `<temporalCoverage>`, check if any elements are in the tile
- for what: extract `<abstract>`, remove prepositions ("of", "at", ...), articles ("the"), ... and check for matches in title
- for where: extract `<geographicDescription>`, cleanup and check for matches 

Use similar files for the other metadata dialects.

**Comment by tedhabermann** [comment]
The target for this check exists in the DataCite Dialect: data[*].attributes.titles[*].title. It is currently implemented as dataset.title.present.
